src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt

```diff
@@ -17,32 +18,41 @@ class PetDaoTest @Autowired constructor (private val sql: DSLContext) {
     private val companyId = Random.nextLong()
     private val petDao = PetDao(sql)
 
+    // Test fields
+    private val ownerId1 = 101L
+    private val ownerId2 = 102L
+    private val petWithoutId1 = PetWithoutId (name = "Rex", type = PetType.DOG, companyId = companyId, dateOfArrival = LocalDate.of(2024,1, 1), ownerId1)
+    private val petWithoutId2 = PetWithoutId (name = "Whiskers", type = PetType.CAT, companyId = companyId, dateOfArrival = LocalDate.of(2024,2,2),ownerId2)
+    private val petWithoutId3 = PetWithoutId(name = "Rexi", type = PetType.DOG, companyId = companyId, dateOfArrival = LocalDate.of(2024,1, 2), ownerId1)
+    private val ownerlessPetWithoutId3 = PetWithoutId(name = "Rexi", type = PetType.DOG, companyId = companyId, dateOfArrival = LocalDate.of(2024,1, 2), null)
+
     @Test
     fun `insert pet test`() {
 
-        petDao.insertPet(name = "Rex", type = PetType.DOG, companyId = companyId, dateOfArrival = LocalDate.of(2024,1, 1))
+        petDao.insertPet(petWithoutId1)
 
         val insertedPet = petDao.getPetsByType(PetType.DOG, companyId)
-        val originPet = Pet(petDao.getPetId(name = "Rex", type = PetType.DOG, companyId = companyId, dateOfArrival = LocalDate.of(2024,1, 1))!!, name = "Rex", type = PetType.DOG, companyId = companyId, dateOfArrival = LocalDate.of(2024,1, 1), null)
+        val petId = petDao.getPetId(petWithoutId1)!!
+        val originPet = Pet(petId, petWithoutId1)
         assertEquals(originPet, insertedPet[0])
     }
 
     @Test
     fun `get pets by type test`() {
 
-        petDao.insertPet(name = "Rex", type = PetType.DOG, companyId = companyId, dateOfArrival = LocalDate.of(2024,1,1))
-        petDao.insertPet(name = "Whiskers", type = PetType.CAT, companyId = companyId, dateOfArrival = LocalDate.of(2024,2,2))
+        petDao.insertPet(petWithoutId1)
```

Comment:
same here 
val dog1Id = petDao.insertPet(petWithoutId1)
val cat2Id = petDao.insertPet(petWithoutId2)

and then this is redundant as you got the id from the isert
val dog1Id = petDao.getPetId(petWithoutId1)
        val cat2Id = petDao.getPetId(petWithoutId2)

Reply:
fixed, amazing feedback